Source/WarpX.cpp

```diff
@@ -1258,13 +1275,22 @@ WarpX::AllocLevelMFs (int lev, const BoxArray& ba, const DistributionMapping& dm
     {
         // Create aux multifabs on Nodal Box Array
         BoxArray const nba = amrex::convert(ba,IntVect::TheNodeVector());
+
         Bfield_aux[lev][0] = std::make_unique<MultiFab>(nba,dm,ncomps,ngE,tag("Bfield_aux[x]"));
         Bfield_aux[lev][1] = std::make_unique<MultiFab>(nba,dm,ncomps,ngE,tag("Bfield_aux[y]"));
         Bfield_aux[lev][2] = std::make_unique<MultiFab>(nba,dm,ncomps,ngE,tag("Bfield_aux[z]"));
 
         Efield_aux[lev][0] = std::make_unique<MultiFab>(nba,dm,ncomps,ngE,tag("Efield_aux[x]"));
         Efield_aux[lev][1] = std::make_unique<MultiFab>(nba,dm,ncomps,ngE,tag("Efield_aux[y]"));
         Efield_aux[lev][2] = std::make_unique<MultiFab>(nba,dm,ncomps,ngE,tag("Efield_aux[z]"));
+
+        Efield_avg_aux[lev][0] = std::make_unique<MultiFab>(*Efield_aux[lev][0], amrex::make_alias, 0, ncomps);
+        Efield_avg_aux[lev][1] = std::make_unique<MultiFab>(*Efield_aux[lev][1], amrex::make_alias, 0, ncomps);
+        Efield_avg_aux[lev][2] = std::make_unique<MultiFab>(*Efield_aux[lev][2], amrex::make_alias, 0, ncomps);
+
+        Bfield_avg_aux[lev][0] = std::make_unique<MultiFab>(*Bfield_aux[lev][0], amrex::make_alias, 0, ncomps);
+        Bfield_avg_aux[lev][1] = std::make_unique<MultiFab>(*Bfield_aux[lev][1], amrex::make_alias, 0, ncomps);
+        Bfield_avg_aux[lev][2] = std::make_unique<MultiFab>(*Bfield_aux[lev][2], amrex::make_alias, 0, ncomps);
```

Comment:
The `aux` and `avg_aux` MultiFabs share the same data (no copies nor updates of any sort needed).

Reply:
In theory yes, we could use directly the `aux` MultiFabs.

The reason why I defined the `avg_aux` MultiFabs is to have an implementation that looks more similar to the energy-conserving case, where we do use the nomenclature `avg_aux` (even if that is simply an alias for `avg_fp`).

Moreover, if you look at this call
https://github.com/ECP-WarpX/WarpX/blob/3f3d60847d18f96a47885a8d5ac3b0c81730d1fe/Source/Evolve/WarpXEvolve.cpp#L622-L626
which then leads to this selection
https://github.com/ECP-WarpX/WarpX/blob/3f3d60847d18f96a47885a8d5ac3b0c81730d1fe/Source/Particles/PhysicalParticleContainer.cpp#L998-L1003
right before gathering the fields, you see that the `aux` and `avg_aux` MultiFabs are passed separately.

If we want to use only the `aux` MultiFabs in case of momentum-conserving gathering, the first call should look like
```cpp
mypc->Evolve(lev, 
             *Efield_aux[lev][0],*Efield_aux[lev][1],*Efield_aux[lev][2], 
             *Bfield_aux[lev][0],*Bfield_aux[lev][1],*Bfield_aux[lev][2], 
             *Efield_aux[lev][0],*Efield_aux[lev][1],*Efield_aux[lev][2], 
             *Bfield_aux[lev][0],*Bfield_aux[lev][1],*Bfield_aux[lev][2],
```
only in the case of momentum-conserving gathering, while it should look like the first one above in the case of energy-conserving gathering.

Maybe we can discuss more on this directly on Slack.